business/checkers/authorization/namespace_method_checker.go

```diff
@@ -0,0 +1,140 @@
+	rules := reflect.ValueOf(rulesStct)
+	if rules.Kind() != reflect.Slice {
+		return checks, valid
+	}
+
+	for ruleIdx := 0; ruleIdx < rules.Len(); ruleIdx++ {
+		rule, ok := rules.Index(ruleIdx).Interface().(map[string]interface{})
+		if !ok || rule == nil {
+			continue
+		}
+
+		if rule["from"] != nil {
+			fromChecks, fromValid := ap.validateFromField(ruleIdx, rule["from"])
+			checks = append(checks, fromChecks...)
+			valid = valid && fromValid
+		}
+
+		if rule["to"] != nil {
+			fromChecks, fromValid := ap.validateToField(ruleIdx, rule["to"])
+			checks = append(checks, fromChecks...)
+			valid = valid && fromValid
+		}
+
+	}
+
+	return checks, valid
+}
+
+func (ap NamespaceMethodChecker) validateFromField(ruleIdx int, from interface{}) ([]*models.IstioCheck, bool) {
+	fromSl, ok := from.([]interface{})
+	if !ok {
+		return nil, true
+	}
+
+	checks, valid := make([]*models.IstioCheck, 0, len(fromSl)), true
+	for fromIdx, fromStc := range fromSl {
+		fromMap, ok := fromStc.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		sourceMap, ok := fromMap["source"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		nsList, ok := sourceMap["namespaces"].([]interface{})
+		if !ok {
+			continue
+		}
+
+		for i, n := range nsList {
+			if !ap.Namespaces.Includes(n.(string)) {
+				valid = true
+				path := fmt.Sprintf("spec/rules[%d]/from[%d]/source/namespaces[%d]", ruleIdx, fromIdx, i)
+				validation := models.Build("authorizationpolicy.source.namespacenotfound", path)
+				checks = append(checks, &validation)
+			}
+		}
+	}
+
+	return checks, valid
+}
+
+func (ap NamespaceMethodChecker) validateToField(ruleIdx int, to interface{}) ([]*models.IstioCheck, bool) {
+	toSl, ok := to.([]interface{})
+	if !ok {
+		return nil, true
+	}
+
+	checks, valid := make([]*models.IstioCheck, 0, len(toSl)), true
+	for toIdx, toStc := range toSl {
+		toMap, ok := toStc.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		sourceMap, ok := toMap["operation"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		mthdList, ok := sourceMap["methods"].([]interface{})
+		if !ok {
+			continue
+		}
+
+		for i, m := range mthdList {
+			if !validMethod(m.(string)) {
+				valid = true
+				path := fmt.Sprintf("spec/rules[%d]/to[%d]/operation/methods[%d]", ruleIdx, toIdx, i)
+				validation := models.Build("authorizationpolicy.to.wrongmethod", path)
+				checks = append(checks, &validation)
+			}
+		}
+	}
+
+	return checks, valid
+}
+
+func validMethod(m string) bool {
+	valid := false
+	httpMethods := []string{http.MethodGet, http.MethodHead, http.MethodPost, http.MethodPut, http.MethodPatch,
+		http.MethodDelete, http.MethodConnect, http.MethodOptions, http.MethodTrace}
+
+	for _, httpMethod := range httpMethods {
+		// HTTP methods allowed or
+		// For gRPC service, a fully-qualified name like “/package.service/method”
+		valid = valid || (strings.TrimSpace(strings.ToUpper(m)) == httpMethod || strings.Contains(m, "/"))
```

Comment:
The logic to check "/package.service/method" format with only strings.Contains(m, "/") looks a little bit weak.
I'd suggest if a regex could be use here instead.
You know, "//////" would be valid according this.

Reply:
More than right. Regexp added.